src/modes/shop/shop_trade.cpp

```diff
@@ -265,6 +265,8 @@ void TradeInterface::MakeActive()
 
 void TradeInterface::TransactionNotification()
 {
+    Reinitialize();
```

Comment:
I added a reinitialize call to the trade and buy interfaces.  Some data structures became out of sync since memory is now being cleaned up.

Note: The sell interface already had this function call.